docs/source/traits_user_manual/internals.rst

```diff
@@ -0,0 +1,190 @@
+.. index:: internals
+
+Traits internals
+================
+
+This section of the documentation is intended both for those developing Traits
+itself, and for those using Traits who would like a better understanding of
+the Traits internals.
+
+
+Fundamentals
+------------
+
+The runtime behavior of Traits is governed by two key classes: |CTrait| and
+|HasTraits|. Each of these classes inherits from a superclass implemented in
+the |ctraits| extension module: |CTrait| inherits from |cTrait|, while
```

Comment:
Nit: "`ctraits` extension module" -> "`ctraits` C extension module" or "`ctraits` module implemented in C"?

Reply:
Fixed.